addons/account/tests/test_transfer_wizard.py

```diff
@@ -445,3 +462,50 @@ def test_transfer_wizard_amount_currency_is_zero(self):
         self.assertRecordValues(destination_line, [
               {'account_id': self.accounts[0].id, 'amount_currency': 0.0, 'currency_id': self.test_currency_1.id, 'balance': -1000}
         ])
+
+    def test_transfer_wizard_analytic(self):
+        """ Tests that the analytic distribution is transmitted when doing a transfer with the wizard """
+        invoice = self.env['account.move'].create([
+            {
+                'move_type': 'out_invoice',
+                'partner_id': self.partner_a.id,
+                'invoice_date': '2017-01-01',
+                'journal_id': self.company_data['default_journal_sale'].id,
+                'invoice_line_ids': [
+                    Command.create({
+                        'quantity': 1,
+                        'price_unit': 1000.0,
+                        'account_id': self.company_data['default_account_revenue'].id,
+                        'analytic_distribution': {self.analytic_account_1.id: 100},
+                    }),
+                    Command.create({
+                        'quantity': 1,
+                        'price_unit': 2000.0,
+                        'account_id': self.company_data['default_account_revenue'].id,
+                        'analytic_distribution': {self.analytic_account_1.id: 50, self.analytic_account_2.id: 50},
+                    }),
+                    Command.create({
+                        'quantity': 1,
+                        'price_unit': 1000.0,
+                        'account_id': self.company_data['default_account_revenue'].id,
+                        'analytic_distribution': False,
+                    }),
+                ],
+            }
+        ])
+        invoice.action_post()
+        wizard = self.env['account.automatic.entry.wizard'].with_context(active_model='account.move.line', active_ids=invoice.invoice_line_ids.ids).create({
+            'action': 'change_account',
+            'date': '2018-01-01',
+            'journal_id': self.journal.id,
+            'destination_account_id': self.receivable_account.id,
+        })
```

Comment:
(cosmetic suggestion)
```suggestion
        wizard = self.env['account.automatic.entry.wizard']\
            .with_context(active_model='account.move.line', active_ids=invoice.invoice_line_ids.ids)\
            .create({
                'action': 'change_account',
                'date': '2018-01-01',
                'journal_id': self.journal.id,
                'destination_account_id': self.receivable_account.id,
            })
```

Reply:
I instead split the context on several lines, I prefer it.